x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorTemplateRegistry.java

```diff
@@ -58,51 +46,8 @@ public class ConnectorTemplateRegistry extends IndexTemplateRegistry {
     // Variable used to replace template version in index templates
     public static final String TEMPLATE_VERSION_VARIABLE = "xpack.application.connector.template.version";
 
-    private static final String MAPPINGS_SUFFIX = "-mappings";
-
-    private static final String SETTINGS_SUFFIX = "-settings";
-
-    private static final String JSON_EXTENSION = ".json";
-
-    static final Map<String, ComponentTemplate> COMPONENT_TEMPLATES;
-
-    static {
-        final Map<String, ComponentTemplate> componentTemplates = new HashMap<>();
-        for (IndexTemplateConfig config : List.of(
-            new IndexTemplateConfig(
-                CONNECTOR_TEMPLATE_NAME + MAPPINGS_SUFFIX,
-                ROOT_TEMPLATE_RESOURCE_PATH + CONNECTOR_TEMPLATE_NAME + MAPPINGS_SUFFIX + JSON_EXTENSION,
-                REGISTRY_VERSION,
-                TEMPLATE_VERSION_VARIABLE
-            ),
-            new IndexTemplateConfig(
-                CONNECTOR_TEMPLATE_NAME + SETTINGS_SUFFIX,
-                ROOT_TEMPLATE_RESOURCE_PATH + CONNECTOR_TEMPLATE_NAME + SETTINGS_SUFFIX + JSON_EXTENSION,
-                REGISTRY_VERSION,
-                TEMPLATE_VERSION_VARIABLE
-            ),
-            new IndexTemplateConfig(
-                CONNECTOR_SYNC_JOBS_TEMPLATE_NAME + MAPPINGS_SUFFIX,
-                ROOT_TEMPLATE_RESOURCE_PATH + CONNECTOR_SYNC_JOBS_TEMPLATE_NAME + MAPPINGS_SUFFIX + JSON_EXTENSION,
-                REGISTRY_VERSION,
-                TEMPLATE_VERSION_VARIABLE
-            ),
-            new IndexTemplateConfig(
-                CONNECTOR_SYNC_JOBS_TEMPLATE_NAME + SETTINGS_SUFFIX,
-                ROOT_TEMPLATE_RESOURCE_PATH + CONNECTOR_TEMPLATE_NAME + SETTINGS_SUFFIX + JSON_EXTENSION,
-                REGISTRY_VERSION,
-                TEMPLATE_VERSION_VARIABLE
-            )
-        )) {
-
-            try (var parser = JsonXContent.jsonXContent.createParser(XContentParserConfiguration.EMPTY, config.loadBytes())) {
-                componentTemplates.put(config.getTemplateName(), ComponentTemplate.parse(parser));
-            } catch (IOException e) {
-                throw new AssertionError(e);
-            }
-        }
-        COMPONENT_TEMPLATES = Map.copyOf(componentTemplates);
-    }
+    // Sources allowed to access system indices using X-elastic-product-origin header
+    public static final List<String> CONNECTORS_ALLOWED_PRODUCT_ORIGINS = List.of("kibana", "connectors");
```

Comment:
Nitpick, can we use origin names defined in `ClientHelper.java` and `KibanaPlugin.java` here instead of hard coding strings?

Reply:
While we're here, @kderusso do you know if Cloud admin portal has a special product origin? I want to make sure we don't lose query access from the admin Elasticsearch API Console

Reply:
@seanstory I'm not sure, I'd suggest reaching out to the team that manages it to verify.

Reply:
I'm following up with the cloud team, if they respond in time for this PR I'll include any necessary changes, otherwise I'll make a follow-up PR.

Reply:
They got back to us, apparently `"cloud"` is the appropriate origin header, so I've added that.